### DISCUSSION TITLE
Creating Non-Optional One-To-One Relationships?

### DISCUSSION THREAD

**Discussion by NayamAmarshe** [prisma/prisma#15816] [Q&A] [answered]
I have a User model in my schema. I want every User to have another table that keeps track of their extra details, like for example their total logins.

So If a User row is created for the first time, the UserDetails table sets the totalLogins to 0 by default.

But the issue is I'm not able to define a `one and only one` relation between both as Prisma throws errors.

What I currently have is this:

```prisma
model User {
  id            String    @id @default(cuid())
  name          String?
  email         String?   @unique
  userDetails   UserDetails?
}

model UserDetails {
  id String     @id      @default(uuid())
  totalLogs     Int      @default(0)
  user          User     @relation(fields: [userId], references: [id], onDelete: Cascade)
  userId        String   @unique
}
```

So basically a relation between User and UserDetails is automatically created on every User creation. Is it possible? Currently UserDetails is not created when User is created.

**Comment by nurul3101** [comment, marked as answer]
Hello @NayamAmarshe 👋 
Based on the shared schema file, I was able to create a user and there was no related userDetails record associated with it.

```ts
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient({
  log: ['query', 'info', 'warn'],
});

async function main() {
  await prisma.user.create({
    data: {
      email: 'test@prisma.io',
      name: 'Test',
    },
  });
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
```
In this case userDetails is not created when a user is created.

<img width="962" alt="Screenshot 2022-10-18 at 7 54 08 PM" src="https://user-images.githubusercontent.com/13049676/196457990-b626b6ce-41c9-4bc6-bc18-6978551bdd23.png">

**Reply by NayamAmarshe** [reply]
> In this case userDetails is not created when a user is created.

Yeah, that's what I'm saying. I would like for it to automatically create UserDetails on User creation but it doesn't :(

**Reply by nurul3101** [reply]
Oh, it won't be created automatically, you would need to do something like this:
```ts
  await prisma.user.create({
    data: {
      email: 'test@prisma.io',
      name: 'Test',
      userDetails: {
        create: {
          totalLogs: 0,
        },
      },
    },
  });
```

**Reply by pantharshit00** [reply]
👋 Hello @NayamAmarshe,  did you perhaps have a chance to check @nurul3101's recommendation? Let us know if you still have any questions!
 
 Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!
